### ISSUE TITLE
Cannot call method 'replace' of undefined in 2.6.6

### ISSUE THREAD

**Issue by algiecas** [vincentbernat/dashkiosk#40]
Dashkiosk v2.6.6 started crashing periodically (once or twice per week). Such error didn't happen in earlier versions.

```
error: uncaughtException: Cannot call method 'replace' of undefined
{ date: 'Mon Feb 08 2016 16:14:45 GMT+0200 (EET)',
process:
{ pid: 16512,
uid: 502,
gid: 20,
cwd: '/Users/macuser/Documents/dashkiosk-2.6.6/dist',
execPath: '/Users/macuser/n/bin/node',
version: 'v0.10.40',
argv:
[ 'node',
'/Users/macuser/Documents/dashkiosk-2.6.6/dist/server.js',
'--environment',
'production',
'--chromecast.enabled',
'--chromecast.receiver',
'http://192,168.0.130:9400/receiver',
'--port',
'9400' ],
memoryUsage:
{ rss: 109051904,
heapTotal: 85364544,
heapUsed: 41827216 } },
os:
{ loadavg:
[ 1.3525390625,
1.3330078125,
1.322265625 ],
uptime: 1060076 },
trace:
[ { column: 13,
file: '/Users/macuser/Documents/dashkiosk-2.6.6/dist/lib/api/socketio/displays.js',
function: '',
line: 17,
method: null,
native: false },
{ column: 17,
file: 'events.js',
function: 'Namespace.emit',
line: 95,
method: 'emit',
native: false },
{ column: 10,
file: '/Users/macuser/Documents/dashkiosk-2.6.6/dist/node_modules/socket.io/lib/namespace.js',
function: 'Namespace.emit',
line: 205,
method: 'emit',
native: false },
{ column: 14,
file: '/Users/macuser/Documents/dashkiosk-2.6.6/dist/node_modules/socket.io/lib/namespace.js',
function: null,
line: 172,
method: null,
native: false },
{ column: 13,
file: 'node.js',
function: 'process._tickDomainCallback',
line: 492,
method: '_tickDomainCallback',
native: false } ],
stack:
[ 'TypeError: Cannot call method \'replace\' of undefined',
' at Namespace. (/Users/macuser/Documents/dashkiosk-2.6.6/dist/lib/api/socketio/displays.js:17:13)',
' at Namespace.emit (events.js:95:17)',
' at Namespace.emit (/Users/macuser/Documents/dashkiosk-2.6.6/dist/node_modules/socket.io/lib/namespace.js:205:10)',
' at /Users/macuser/Documents/dashkiosk-2.6.6/dist/node_modules/socket.io/lib/namespace.js:172:14',
' at process._tickDomainCallback (node.js:492:13)' ] }
```

**Comment by vincentbernat** [comment]
Hi!

Which version of socket.io did you get? `npm list --depth 0 | grep socket.io` should tell you the version.

**Comment by algiecas** [comment]
`socket.io@1.3.7`

**Comment by vincentbernat** [comment]
OK, it's recent enough. It seems that on some transports, the remote address is not available. There are multiple reports about that but it's hard to know when this happens and if it should still happen. I have committed a fix for those cases (ca98006).
